### ISSUE TITLE
MENSTART format in menstrual cycle survey should be MM/DD/YYYY (not YYYY-MM-DD)

### ISSUE THREAD

**Issue by cusackjm** [episphere/questionnaire#166]
In the biospecimens qx MENSTART is formatted as MM/DD/YYYY, but it is piped into the menstrual cycle survey as YYYY-MM-DD 
![image](https://user-images.githubusercontent.com/85582895/185453539-08cf43c4-299a-4e8c-a4f4-7887e8c488a8.png)

Could you change the format for how the date is piped into the menstrual cycle survey?

**This is a low priority "later" problem**

**Comment by anthonypetersen** [comment]
@FrogGirl1123 is there any reason we can't just update the expected format in the Data Dictionary for the affected variables? (644459734, 593467240)

If not, this will require a large number of manual updates to existing data and will also require a change in Quest that will affect the output of the `date` question type. This question type has existed in its current format for quite some time and is likely being used by people outside of Connect.

**Comment by cunnaneaq** [comment]
Thanks for looking into this @anthonypetersen!

Sounds good, @FrogGirl1123!  I agree the "Current Format/Value" in the DD is MM/DD/YYYY and this was likely posted as an issue because YYYY-MM-DD is less familiar.  This can wait until MCS v1.2.

**Comment by danielruss** [comment]
@FrogGirl1123  The [html standard for the date picker](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date) is YYYY-MM-DD.  I thought when the data was passed to firestore, the data had the format conversion.  The issue we ran into was if you take `new Date("2021-04-03")` and then look at the value it is `Fri Apr 02 2021 20:00:00 GMT-0400 (Eastern Daylight Time)`.  The Date in JS is actually a date-time of midnight UTC.  In the conversion to EDT the date rolls back one day.

**Comment by FrogGirl1123** [comment]
@danielruss so what it sounds like we really need is something in the PWA or Firestore that delivers the date as mmddyyyy when it's piped back to the participant so that there's no confusion with month vs day.

**Comment by danielruss** [comment]
Is the problem the display or is the problem the stored data.

If the problem is the stored data, can the process that moves the data from firestore to BQ handle the conversion?
Otherwise, see if we can modify quest to make sure date are formatted outside of `<input type="date">` as MM/DD/YYYY.

**Comment by anthonypetersen** [comment]
I don't think that is a valid solution. The markdown is grabbing the variable as it exists in Firestore, I don't want us building one-off code to check for certain variables to display them differently than they are stored.

If we want this fixed we should be pursuing a solution that stores data in the correct format.

**Comment by anthonypetersen** [comment]
I'd also prefer that our backend have to parse every Quest response to see if we need to change the format. My idea for solving this would be to expand upon the `|date|` functionality in Quest to allow users to define how they want the user input formatted. Thoughts? @danielruss

**Comment by danielruss** [comment]
The markdown in question is  {$u:D_644459734} .  

This is replaced with math._value() It is possible to change this in the _value() function.

check if it matches /\d{4}-\d{2}-\{2}/. if so reformat and return....

**Comment by FrogGirl1123** [comment]
Maybe we could discuss this on a Tuesday DevOps support call.

**Comment by danielruss** [comment]
see episphere/quest#461
